src/api/handlers/supported_language.py

```diff
@@ -0,0 +1,62 @@
+from _main_.utils.constants import INVALID_LANGUAGE_CODE_ERR_MSG
+from _main_.utils.route_handler import RouteHandler
+from _main_.utils.massenergize_response import MassenergizeResponse
+from _main_.utils.context import Context
+from api.decorators import super_admins_only
+from api.services.supported_language import SupportedLanguageService
+
+
+class SupportedLanguageHandler(RouteHandler):
+
+    def __init__(self):
+        super().__init__()
+        self.service = SupportedLanguageService()
+        self.registerRoutes()
+
+    def registerRoutes(self):
+        self.add("/supported_languages.info", self.info)
+        self.add("/supported_languages.create", self.create)
+        self.add("/supported_languages.add", self.create)
+        self.add("/supported_languages.list", self.list)
+
+    def info(self, request):
+        context: Context = request.context
+        args: dict = context.args
+
+        code = args.get('language_code', None)
+
+        if not code:
+            return MassenergizeResponse(error= INVALID_LANGUAGE_CODE_ERR_MSG)
+
+        supported_language_info, err = self.service.get_supported_language_info(context, args)
+        if err:
+            return err
+
+        if not supported_language_info:
+            return MassenergizeResponse(error="No supported language found with the provided code")
+        return MassenergizeResponse(data=supported_language_info)
+
+    @super_admins_only
+    def create(self, request):
+        context: Context = request.context
+        args = context.get_request_body()
+
+        (self.validator
+         .expect("language_code", str, is_required=True)
+         .expect("name", str, is_required=True)
+         )
+        args, err = self.validator.verify(args)
+
+        if err:
+            return MassenergizeResponse(error="Please provide a valid language code and name")
```

Comment:
you can even define the error messages texts at the top of the file as constants.